### ISSUE TITLE
Executor idle timer not reset

### ISSUE THREAD

**Issue by macintoshpie** [Parsl/parsl#782]
TLDR; I think there's a bug in assessing timeout for idle executors.

Parsl log: https://gist.github.com/macintoshpie/1ae87cd93e45e532513ea2833a3f8e60

### Setup
Running sequence of 5 tasks that each last about 3 hours on NSCC Aspire1, on a single executor.

### Problem
The first task completes after about 3 hours but then the executor is killed and the parsl head errors out.
See this line for where things go awry: https://gist.github.com/macintoshpie/1ae87cd93e45e532513ea2833a3f8e60#file-parsl-log-L5197

### Proposed bug
You can see in [this line](https://gist.github.com/macintoshpie/1ae87cd93e45e532513ea2833a3f8e60#file-parsl-log-L88) that the `idle_time` is set early in the process. There's no logging that the kill timer is being reset/updated later on either. Once the first task finishes, it meets the conditions to enter a potential timeout state (`active_tasks == 0 && active_blocks > min_blocks`). It looks at the very old `idle_time` set earlier and kills the executor immediately.

The solution is to reset the timer once it's no longer in the possible timeout state. Here's the problematic area of code: https://github.com/Parsl/parsl/blob/b4c55567b468278e15516e89373bc0a72d445317/parsl/dataflow/strategy.py#L209-L228
